### PR TITLE
fix OverflowException on Win11 x64

### PR DESCRIPTION
**Pull request by KenMercusLai**
Fix the exception when running on Windows 11 x64:
Unhandled exception. System.OverflowException: Arithmetic operation resulted in an overflow.
   at System.IntPtr.op_Explicit(IntPtr value)
   at workspacer.ConsoleHelper.get_IsConsoleShowing() in D:\a\workspacer\workspacer\src\workspacer.Watcher\ConsoleHelper.cs:line 35
   at workspacer.ConsoleHelper.set_IsConsoleShowing(Boolean value) in D:\a\workspacer\workspacer\src\workspacer.Watcher\ConsoleHelper.cs:line 46
   at workspacer.ConsoleHelper.Initialize() in D:\a\workspacer\workspacer\src\workspacer.Watcher\ConsoleHelper.cs:line 25
   at workspacer.Watcher.Program.Main(String[] args) in D:\a\workspacer\workspacer\src\workspacer.Watcher\Program.cs:line 19

### CODE REVIEW

File: src/workspacer.Watcher/ConsoleHelper.cs [hunk 1]
```diff
@@ -27,6 +27,9 @@ public static bool IsConsoleShowing
                 if (handle != IntPtr.Zero)
                 {
                     var style = Win32.GetWindowLongPtr(handle, Win32.GWL_STYLE);
+                    if (Environment.Is64BitProcess) {
+                        return ((UInt64)style & (UInt64)Win32.WS.WS_VISIBLE) != 0;
+                    }
```
Comment (josteink) [inline src/workspacer.Watcher/ConsoleHelper.cs:32]: This fixes the getter code. Do we need to fix the setter code too?

**Comment by josteink** [comment]
Hey there and thanks for the patch!

First of all: Sorry for the slow response.

 If you can look into my 1 single feedback, I'll do my best to get this fix merged and mainlined ASAP :smile:

File: src/workspacer.Watcher/ConsoleHelper.cs (hunk 1 above)
Comment (KenMercusLai) [reply src/workspacer.Watcher/ConsoleHelper.cs:32]: I think we need, but I haven't caught any error on that part. 😄

Most awkward part is I don't know how to write C#. I fixed the issue by googling. 😂

**Comment by KenMercusLai** [comment]
No worries mate. I'm playing with workspacer to learn more about it.

File: src/workspacer.Watcher/ConsoleHelper.cs (hunk 1 above)
Comment (josteink) [reply src/workspacer.Watcher/ConsoleHelper.cs:32]: So... Should I interpret that to mean you won't be able to fix this? And that someone else will have to help out?

CC: @tlsomers - Can you help out here?

Comment (tlsomers) [reply src/workspacer.Watcher/ConsoleHelper.cs:32]: I see that this functionality has already been implemented in `Win32. GetWindowStyleLongPtr`, why not use that method and the [Enum.HasFlag](https://docs.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=net-6.0) method.

**Review by tlsomers — CHANGES_REQUESTED** [review]

**Comment by josteink** [comment]
Hey there @KenMercusLai . Thanks for bringing this bug to our attention and a partial fix.

@tlsomers has now provided a complete fix for this in another PR, so I'll go ahead and close this one.

If you have any other contributions in the future, don't be afraid to open a new PR :smile:

**Pull request closed without merging**